### PR TITLE
FIX: Fix KNN warning on unused metric parameter during fallbacks

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

Fixes an issue with an unused metric parameter being passed to sklearn during fallbacks, which triggers a warning.

---

<details>
<summary>Checklist:</summary>

**Completeness and readability**

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/doc/sources/contribute.rst) for details)_.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

</details>

### CODE REVIEW

File: sklearnex/neighbors/common.py
Comment (ethanglaser) [inline sklearnex/neighbors/common.py:1]: What is the warning message and how does this change address it?

Comment (david-cortes-intel) [reply sklearnex/neighbors/common.py:1]: Warning is:
```
<path>/sklearnex/_device_offload.py:172: UserWarning: Some metric_kwargs have been passed ({'p': 2}) but aren't usable for this case (EuclideanArgKmin64) and will be ignored.
```

It's pretty self explanatory.

**Review by yuejiaointel — APPROVED** [review]
LGTM!
